java-runtimes-common/test-spring-application/pom.xml

```diff
@@ -0,0 +1,153 @@
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.build.directory}</outputDirectory>
+              <resources>
+                <resource>
+                  <directory>${basedir}/src/main/appengine</directory>
+                  <filtering>false</filtering>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>com.coderplus.maven.plugins</groupId>
+        <artifactId>copy-rename-maven-plugin</artifactId>
+        <version>1.0.1</version>
+        <executions>
+          <execution>
+            <id>rename-file</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>rename</goal>
+            </goals>
+            <configuration>
+              <sourceFile>${project.build.directory}/${app.yaml.file}</sourceFile>
+              <destinationFile>${project.build.directory}/deploy/app.yaml</destinationFile>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <!-- this profile stages the application for integration tests -->
+      <id>int-test</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <properties>
+        <app.yaml.file>app.int-test.yaml</app.yaml.file>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-resources-plugin</artifactId>
+            <version>3.0.2</version>
+            <executions>
+              <execution>
+                <id>copy-resources</id>
+                <phase>prepare-package</phase>
+                <goals>
+                  <goal>copy-resources</goal>
+                </goals>
+                <configuration>
+                  <outputDirectory>${project.build.directory}/deploy</outputDirectory>
+                  <resources>
+                    <resource>
+                      <directory>${basedir}/src/main/docker</directory>
+                      <filtering>false</filtering>
+                    </resource>
+                    <resource>
+                      <directory>${basedir}/src/main/kubernetes</directory>
+                      <filtering>false</filtering>
+                    </resource>
+                  </resources>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <!-- this profile stages the application for deployment tests -->
+      <id>deployment-test</id>
+      <properties>
+        <app.yaml.file>app.deploy-check.yaml</app.yaml.file>
```

Comment:
Are these two profiles (deployment-test and int-test) even warranted, when the only difference is the `app.yaml.file` property? Also, I still find it confusing in terms of naming because both of these profiles support integration tests. It's just that one is local and another is remote, or am I missing something?